spyder/app/mainwindow.py

```diff
@@ -567,7 +569,24 @@ def create_toolbar(self, title, object_name, iconsize=24):
         self.toolbarslist.append(toolbar)
         return toolbar
 
-    def setup(self):
+    def setup_menus(self):
+        """Hide icon menus in toolbars before displaying them"""
+
+        if sys.platform == 'darwin':
+
+            self.file_menu.aboutToShow.connect(lambda menu_actions=self.file_menu_actions: self.verify_menu_actions(menu_actions, False))

```

Comment:
@ccordoba12 could we add toolbars to a variable upon creation so that we can iterate over that variable?

Your suggestion also works, but I think it would be easier to keep track of all menus in a single variable

Thoughts?

Reply:
Sure, that's fine too.